test/unit/org/apache/cassandra/cql3/CQLTester.java

```diff
@@ -1933,6 +1958,59 @@ private static boolean isEmptyContainerNull(DataType type,
         return false;
     }
 
+    /**
+     * Determine whether the source and target TableMetadata is equal without compare the table name and dropped columns.
+     * @param source the source TableMetadata
+     * @param target the target TableMetadata
+     * @param compareParams wether compare table params
+     * @param compareIndexes wether compare table's indexes
+     * @param compareTrigger wether compare table's triggers
+     * */
+    protected boolean equalsWithoutTableNameAndDropCns(TableMetadata source, TableMetadata target, boolean compareParams, boolean compareIndexes, boolean compareTrigger)
+    {
+        return source.partitioner.equals(target.partitioner)
+               && source.kind == target.kind
+               && source.flags.equals(target.flags)
+               && (!compareParams || source.params.equals(target.params))
+               && (!compareIndexes || source.indexes.equals(target.indexes))
+               && (!compareTrigger || source.triggers.equals(target.triggers))
+               && columnsEqualWitoutKsTb(source, target);
+    }
+
+    // only compare columns
+    private boolean columnsEqualWitoutKsTb(TableMetadata source, TableMetadata target)
+    {
+        if (target.columns() == source.columns())
+            return true;
+
+        List<ColumnMetadata> left = source.columns().stream().sorted().collect(Collectors.toList());
+        List<ColumnMetadata> right = target.columns().stream().sorted().collect(Collectors.toList());
+
+        if (left.size() != right.size())
+            return false;
```

Comment:
If we are going to use the iterator anyway we should use TableMetadata::allColumnsInCreateOrder and instead of comparing the size checking that both iterator are exhausted at the end.